### DISCUSSION TITLE
How do I get the underlying markdown list number for an ordered list (ol)?

### DISCUSSION THREAD

**Discussion by nick-inkeep** [remarkjs/remark#1139] [Q&A] [answered]
I have markdown (that I don't control) that can be in the following format:

```
# Heading
3. My name
4. My age
```

I would like to show in the HTML the same number as is in the markdown, i.e. "3." and "4.". Currently, it resets to zero for every ol. I want to respect the markdown ordering.

**Comment by ChristianMurphy** [comment, marked as answer]
Markdown numbers based off the first item in the list.
In your case 3.

If you are seeing a different number, most likely a configuration option, stylesheet, or plugin is modifying your lists.

**Comment by nick-inkeep** [comment]
is there a way to get that number in the renderer? I'd like to customize the rendered component a bit and use my ui-library. 'start' of the default rendered ol doesn't seem to be available for me as a prop in the renderer to pass to my custom component.

E.g.:
```
   ol: ({ children }: { children: React.ReactNode }) => (
            <OrderedList
                as="ol"
                sx={contentParserStyles.ol}
                overflowWrap="break-word"
            >
                {children}
            </OrderedList>
        ),
```

**Reply by wooorm** [reply]
You ignore all props. You only use `children` and drop the rest.
You are being given a prop: See https://github.com/remarkjs/react-markdown#appendix-b-components.

**Comment by vadimkantorov** [comment]
For me, restarting doesn't work:

```markdown
1. hello
2. world


1. foo bar
2. zoo 
```

gets rendered in [remark](https://remark.js.org) as:

```
1. hello
2. world
3. foo bar
4. zoo
```

**Reply by remcohaszing** [reply]
This is how markdown works. Try pasting your example in GitHub for example.
